source_pubsub/position/position.go

```diff
@@ -0,0 +1,83 @@
+package position
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+type Topics struct {
+	Topics TopicPositions `json:"topics"`
+}
+
+type TopicPositions map[string]TopicPosition
+
+type TopicPosition struct {
+	ReplayID []byte    `json:"replayID"`
+	ReadTime time.Time `json:"readTime"`
+}
+
+func ParseSDKPosition(sdkPos sdk.Position) (Topics, error) {
+	var p Topics
+	p.Topics = make(TopicPositions)
+
+	if len(sdkPos) == 0 {
+		return p, nil
+	}
+
+	if err := json.Unmarshal(sdkPos, &p); err != nil {
+		return p, fmt.Errorf("invalid position: %w", err)
+	}
+	return p, nil
+}
+
+func NewTopicPosition() Topics {
+	var p Topics
+	p.Topics = make(TopicPositions)
+	return p
+}
+
+func (p Topics) SetTopics(topics []string) {
+	for _, topic := range topics {
+		if _, ok := p.Topics[topic]; !ok {
+			replayEvent := TopicPosition{
+				ReplayID: nil,
+			}
+			p.Topics[topic] = replayEvent
+		}
+	}
+}
+
+func (p Topics) GetTopicReplayID(topic string) []byte {
```

Comment:
```suggestion
func (p Topics) TopicReplayID(topic string) []byte {
```

No need for the get.